rlclientlib/federation/local_loop_controller.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include "api_status.h"
+#include "error_callback_fn.h"
+#include "factory_resolver.h"
+#include "federation/event_sink.h"
+#include "federation/federated_client.h"
+#include "federation/joined_log_provider.h"
+#include "federation/vw_trainable_model.h"
+#include "model_mgmt.h"
+#include "sender.h"
+#include "trace_logger.h"
+
+#include <functional>
+#include <memory>
+
+namespace reinforcement_learning
+{
+// The local_loop_controller will "plug in" to rlclientlib as an i_data_transport object.
+// It exposes a get_local_sender_factory() function that creates i_sender proxy objects.
+// These proxy objects will send events to its internal event sink.
+// The initialization code for live_model_impl must register this factory function correctly.
+class local_loop_controller : public model_management::i_data_transport
+{
+public:
+  RL_ATTR(nodiscard)
+  static int create(std::unique_ptr<local_loop_controller>& output, const utility::configuration& config,
+      i_trace* trace_logger = nullptr, api_status* status = nullptr);
+
+  // Get model data in binary format
+  // This will perform joining and training on any observed events, and then return the updated model
+  RL_ATTR(nodiscard)
+  virtual int get_data(model_management::model_data& data, api_status* status = nullptr) override;
```

Comment:
if this will perform joining and training and then return the updated model it should have a name that indicates that it does so

Reply:
never mind, I see you are overriding `i_data_transport`